x/staking/keeper/cons_pubkey.go

```diff
@@ -48,6 +49,42 @@ func (k Keeper) setConsPubKeyRotationHistory(
 	return k.setConsKeyQueue(ctx, queueTime, valAddr)
 }
 
+func (k Keeper) updateToNewPubkey(ctx sdk.Context, val types.Validator, oldPubKey, newPubKey *codectypes.Any, fee sdk.Coin) error {
```

Comment:
some more comments here would be nice, it's not immediately clear to me why we call a hook